soh/src/code/z_collision_check.c

```diff
@@ -3008,7 +3008,11 @@ void CollisionCheck_ApplyDamage(PlayState* play, CollisionCheckContext* colChkCt
         collider->actor->colChkInfo.damageEffect = tbl->table[i] >> 4 & 0xF;
     }
     if (!(collider->acFlags & AC_HARD)) {
-        collider->actor->colChkInfo.damage += damage;
+        // Decomp:
+        // collider->actor->colChkInfo.damage += damage;
+        // SoH:
+        // Prevents enemies taking double damage when Hyper Bosses is on.
+        collider->actor->colChkInfo.damage = damage;
     }
```

Comment:
So this one will need a bit of explanation. It turns out that running the update function "queues" up the collision check, then runs them all at the same time afterwards. So it runs this code twice, THEN runs the actual actor update function twice. The problem with this is that this function is called twice before the actual "apply damage" functions are called in Bongo Bongo and normal enemies code, causing them to take double damage.

The async nature of this also means I can't reset the collision value back to 0 before running this a second time inside the hyper boss code.

I don't know *why* the vanilla game is using += here. I can't find anything in the code that's making use of some kind of cumulative damage and I've tested Tektites and Bongo Bongo (2 actors using this code to apply damage) and they work just fine with them both while running a second update function and while not running a second update function.

This might seem like a harsh fix for just fixing double damage on Bongo Bongo, but this will apply to pretty much every single normal enemy in the game if we ever want to get Hyper Enemies implemented as well. Technically I could put this behind a CVar, but it'll apply to every actor even if it's just hyper bosses that's enabled, and as far as I can tell this change doesn't actually hurt the vanilla game anyway, but can implement it if we'd still rather be safe than sorry.

Reply:
Disregard the above. With the help of the 3DS peeps (Kewlan and Hylian Freddy), a proper fix that actually targets the root cause is now pushed.